hoomd/md/nlist.py

```diff
@@ -190,235 +175,138 @@ def _detach(self):
-        dist_check (bool): Flag to enable / disable distance checking.
-        cell_width (float): The underlying stencil bin width for the cell list
-        name (str): Optional name for this neighbor list instance.
-        deterministic (bool): When True, enable deterministic runs on the GPU by sorting the cell list.
-
-    :py:class:`stencil` creates a cell list based neighbor list object to which pair potentials can be attached for computing
-    non-bonded pairwise interactions. Cell listing allows for O(N) construction of the neighbor list. Particles are first
-    spatially sorted into cells based on the largest pairwise cutoff radius attached to this instance of the neighbor
-    list.
-
-    `M.P. Howard et al. 2016 <http://dx.doi.org/10.1016/j.cpc.2016.02.003>`_ describes this neighbor list implementation
-    in HOOMD-blue. Cite it if you utilize this neighbor list style in your work.
-
-    This neighbor-list style differs from :py:class:`Cell` based on how the adjacent cells are searched for particles. The cell
-    list *cell_width* is set by default using the shortest active cutoff radius in the system. One *stencil* is computed
-    per particle type based on the largest cutoff radius that type participates in, which defines the bins that the
-    particle must search in. Distances to the bins in the stencil are precomputed so that certain particles can be
-    quickly excluded from the neighbor list, leading to improved performance compared to :py:class:`Cell` when there is size
-    disparity in the cutoff radius. The memory demands of :py:class:`stencil` can also be lower than :py:class:`Cell` if your
-    system is large and has many small cells in it; however, :py:class:`tree` is usually a better choice for these systems.
-
-    The performance of the stencil depends strongly on the choice of *cell_width*. The best performance is obtained
-    when the cutoff radii are multiples of the *cell_width*, and when the *cell_width* covers the simulation box with
-    a roughly integer number of cells. The *cell_width* can be set manually, or be automatically scanning through a range
-    of possible bin widths using :py:meth:`tune_cell_width()`.
+        cell_width (float): The underlying stencil bin width for the cell list.
+        buffer (float): Buffer width.
+        exclusions (tuple[str]): Excludes pairs from the neighbor list, which
+            excludes them from the pair potential calculation.
+        rebuild_check_delay (int): How often to attempt to rebuild the neighbor
+            list.
+        diameter_shift (bool): Flag to enable / disable diameter shifting.
+        check_dist (bool): Flag to enable / disable distance checking.
+        max_diameter (float): The maximum diameter a particle will achieve.
+        deterministic (bool): When `True`, sort neighbors to help provide
+            deterministic simulation runs.
```

Comment:
```suggestion
```
I don't see `deterministic` in the C++ code for the Stencil neighbor list. I don't think it supports this option.

Reply:
After taking a closer look, I think this is something I forgot to write get/setDeterministic methods for and export in the NeighborListStencil.

Reply:
> After taking a closer look, I think this is something I forgot to write get/setDeterministic methods for and export in the NeighborListStencil.

I may not be remembering right, but I feel like I did not implement this option for the stencil neighbor list. It would be easy to add it if you wanted to copy the getter / setter from `NeighborListGPUBinned`. I think I just didn't see much of a potential use case at the time. (`deterministic=True` triggers an extra sort of the cell list... I found `NeighborListGPUTree` worked better for most cases where you would use `NeighborListGPUStencil`, and the tree is inherently deterministic too.)

Reply:
On the v2.9 branch, you can look and see that there is a deterministic argument passed to the stencil constructor which is passed through the setSortCellList method of the cell list class.

Reply:
Ah, I see, I must have done this at a purely python level. I only skimmed the C++ code, which does not handle this option. If you want to keep it but remove the external cell list, you would need to add the C++ methods to forward that flag to the cell list.